### PR TITLE
fixed bug

### PR DESCRIPTION
**Pull request by obucklin**
Fixed a bug where the direct joint rule used the beam order input into the assembly component instead of what is specified at the joint options component. 

Also changed the description of the DirectRule class

### What type of change is this?

- [`x` ] Bug fix in a **backwards-compatible** manner.
- [ ] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [ `x`] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [ `x`] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas_timber.datastructures.Beam`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by obucklin** [comment]
Ok so this gets the beams to the joint in the right order, but it seems like something is happening in the `LButtJoint` is switching the order sometimes. will investigate further

**Comment by chenkasirer** [comment]
Yep, also fixed it my PR https://github.com/gramaziokohler/compas_timber/pull/213/files#diff-6771d3911510682c772cf81e12b3b91f0ad2807853fd944612680eeb1b36fb23R61

Yea add it to changelog as it's a bug fix. 

Maybe give https://github.com/gramaziokohler/compas_timber/pull/213 a look and review it. Just so we're solving the same issues in parallel.

File: src/compas_timber/ghpython/components/CT_Assembly/code.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ def get_joints_from_rules(self, beams, rules, topologies):
 
             for rule in direct_rules:  # apply direct rules first
                 if rule.comply(pair):
-                    joints.append(JointDefinition(rule.joint_type, [beam_a, beam_b], **rule.kwargs))
+                    joints.append(JointDefinition(rule.joint_type, [rule.beams[0], rule.beams[1]], **rule.kwargs))
```
Comment (chenkasirer) [inline src/compas_timber/ghpython/components/CT_Assembly/code.py:61]: why not just `rule.beams`?

Comment (chenkasirer) [inline src/compas_timber/ghpython/components/CT_Assembly/code.py:61]:
```suggestion
                    joints.append(JointDefinition(rule.joint_type, rule.beams, **rule.kwargs))
```

**Pull request merged**